src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -632,27 +641,77 @@ export const Deposit: React.FC<DepositProps> = ({
       contextDispatch({ type: ThorchainSaversDepositActionType.SET_LOADING, payload: true })
       setIsSendMax(percent === 1)
 
+      const _percentageCryptoAmountBaseUnitBeforeTxFees = bnOrZero(balanceCryptoBaseUnit)
+        .times(percent)
+        .toFixed(0)
+
       const _percentageCryptoAmountPrecisionBeforeTxFees = bnOrZero(balanceCryptoPrecision)
         .times(percent)
         .dp(asset.precision)
 
+      const assetAddress = !isToken(assetId)
+        ? zeroAddress
+        : getAddress(fromAssetId(assetId).assetReference)
+
+      const thorchainDepositQuoteQueryKey: GetThorchainSaversDepositQuoteQueryKey = [
+        'thorchainSaversDepositQuote',
+        { asset, amountCryptoBaseUnit: _percentageCryptoAmountPrecisionBeforeTxFees },
+      ]
+
+      let inboundAddressRouter, depositInputData
+
+      if (isEvmChainId(asset.chainId)) {
+        const [thorchainDepositQuote, inboundAddresses] = await Promise.all([
+          queryClient.fetchQuery({
+            queryKey: thorchainDepositQuoteQueryKey,
+            queryFn: () =>
+              fetchThorchainDepositQuote({
+                asset,
+                amountCryptoBaseUnit: toBaseUnit(
+                  _percentageCryptoAmountPrecisionBeforeTxFees,
+                  asset.precision,
+                ),
+              }),
+          }),
+          queryClient.fetchQuery({
+            ...reactQueries.thornode.inboundAddresses(),
+          }),
+        ])
+
+        const inboundAddressData = selectInboundAddressData(inboundAddresses, assetId)
+        const inboundAddress = inboundAddressData?.address
+        inboundAddressRouter = inboundAddressData?.router
+
+        depositInputData =
+          feeAsset?.chainId && isEvmChainId(feeAsset.chainId)
```

Comment:
`isEvmChainId(feeAsset.chainId)` will always be true here - this is within a `if (isEvmChainId(asset.chainId)) {}` clause